### PR TITLE
Make Globals writeable via the API

### PR DESCRIPTION
**Pull request by tybritten**
This PR is WIP. What's working:
- `GET` Global using key to return a single global
- `POST` Global using key to change value

What's not:
- `POST` Global using name to create a new global with value
- tests for all of the above

### CODE REVIEW

**Comment by tybritten** [comment]
@nicpottier or @rowanseymour I'd love an initial review to see if I'm going off the rails at all. I'm troubleshooting my create global option currently as it's not working.

File: temba/api/v2/views.py [hunk 1]
```diff
@@ -1984,15 +1986,74 @@ class GlobalsEndpoint(ListAPIMixin, BaseAPIView):
...
+        POST /api/v2/global.json
+        {
+            "name": "Org Name",
+            "value": "Acme Ltd"
+        }
+
+    You will receive a global object as a response if successful:
+
+        {
+        "key": "org_name",
```
Comment (rowanseymour) [inline temba/api/v2/views.py]: indentation

File: temba/api/v2/serializers.py [hunk 2]
```diff
@@ -1043,6 +1043,40 @@ class Meta:
...
+        return value
+
+    def validate(self, data):
+        key = self.context["lookup_values"]["key"]
+        value = data["value"]
+        # make sure this combination doesn't already exist
+        if Global.objects.filter(
+            key=key, value=value, is_active=True
+        ):  # pragma: needs cover
+            raise serializers.ValidationError("This Global is already set to this value")
```
Comment (rowanseymour) [inline temba/api/v2/serializers.py]: don't think this needs to be an error

**Comment by tybritten** [comment]
Ok, I fixed those two and now have Global creation working. I'm just calling `Global.make_key` to create the key from name. Should I be then calling `is_valid_key` too?

File: temba/api/v2/views.py [hunk 3]
```diff
@@ -1984,15 +1986,74 @@ class GlobalsEndpoint(ListAPIMixin, BaseAPIView):
...
+        {
+            "key": "org_name",
+            "name": "Org Name",
+            "value": "Acme Ltd",
+            "modified_on": "2013-02-27T09:06:15.456"
+        }
+
+    ## Updating a Global
+
+    A **POST** can also be used to update an existing contact group if you specify its UUID in the URL.
```
Comment (chris-erickson) [inline temba/api/v2/views.py]: Contact group left over, should be global

File: temba/api/v2/views.py [hunk 4]
```diff
@@ -1984,15 +1986,74 @@ class GlobalsEndpoint(ListAPIMixin, BaseAPIView):
...
+
+    Example:
+
+        POST /api/v2/globals.json?key=org_name
+        {
+            "value": "Acme Ltd"
+        }
+
+    You will receive the updated group object as a response if successful:
```
Comment (chris-erickson) [inline temba/api/v2/views.py]: Group

**Comment by rowanseymour** [comment]
@tybritten yes. 

Also don't forget you can now use `./code_check.py` to run all the code quality checks we run in travis

File: temba/api/v2/serializers.py [hunk 5]
```diff
@@ -1043,6 +1043,33 @@ class Meta:
         fields = ("key", "name", "value", "modified_on")
 
 
+class GlobalWriteSerializer(WriteSerializer):
+    value = serializers.CharField(required=True)
+    name = serializers.CharField(
```
Comment (norkans7) [inline temba/api/v2/serializers.py]: I think `name` should be required

Comment (tybritten) [reply temba/api/v2/serializers.py]: I didn't make it required for the case of updating an existing global- just supply the key and new value. Is there a problem with that approach?

Comment (norkans7) [reply temba/api/v2/serializers.py]: In that case you need a `key` I think and check that either is provided and never both missing
I am not sure how that existing object would be retrieved if we do not provide the key or the name

Comment (norkans7) [reply temba/api/v2/serializers.py]: never mind I got it https://github.com/nyaruka/rapidpro/pull/2699/files#diff-8d599bcb3650f809de7d5aa0f5bb313dR2040

**Comment by rowanseymour** [comment]
@tybritten can you get the coverage fixed on this today or tomorrow so it can be in 5.4?

**Comment by tybritten** [comment]
Wasn’t there coverage before my last commit? I can’t see where we’re missing coverage.

**Comment by rowanseymour** [comment]
@tybritten 

```
Name                          Stmts   Miss  Cover   Missing
-----------------------------------------------------------
temba/api/v2/serializers.py     776      1    99%   1069
temba/api/v2/views.py           654      2    99%   2047, 2053
-----------------------------------------------------------
TOTAL                         23969      3    99%
```

**Comment by tybritten** [comment]
Thanks! I’ll get those covered.

**Pull request merged**